src/Core/Asset/GeometryData.inl

```diff
@@ -188,47 +197,191 @@ inline bool GeometryData::isHexMesh() const {
 }
 
 inline bool GeometryData::hasVertices() const {
-    return !m_vertex.empty();
+    return m_multiIndexedGeometry.vertexAttribs().contains( "in_position" );
 }
 
 inline bool GeometryData::hasEdges() const {
-    return !m_edge.empty();
+    return m_multiIndexedGeometry.containsLayer( { Geometry::LineIndexLayer::staticSemanticName },
+                                                 "in_edge" );
 }
 
 inline bool GeometryData::hasFaces() const {
-    return !m_faces.empty();
+    return m_multiIndexedGeometry.containsLayer( { Geometry::PolyIndexLayer::staticSemanticName },
+                                                 "in_face" );
 }
 
 inline bool GeometryData::hasPolyhedra() const {
-    return !m_polyhedron.empty();
+    return m_multiIndexedGeometry.containsLayer( { Geometry::PolyIndexLayer::staticSemanticName },
+                                                 "in_polyhedron" );
 }
 
 inline bool GeometryData::hasNormals() const {
-    return !m_normal.empty();
+    return m_multiIndexedGeometry.vertexAttribs().contains( "in_normal" );
 }
 
 inline bool GeometryData::hasTangents() const {
-    return !m_tangent.empty();
+    return m_multiIndexedGeometry.vertexAttribs().contains( "in_tangent" );
 }
 
 inline bool GeometryData::hasBiTangents() const {
-    return !m_bitangent.empty();
+    return m_multiIndexedGeometry.vertexAttribs().contains( "in_bitangent" );
 }
 
 inline bool GeometryData::hasTextureCoordinates() const {
-    return !m_texCoord.empty();
+    return m_multiIndexedGeometry.vertexAttribs().contains( "in_texcoord" );
 }
 
 inline bool GeometryData::hasMaterial() const {
     return m_material != nullptr;
 }
 
+const Geometry::MultiIndexedGeometry& GeometryData::getMultiIndexedGeometry() const {
+    return m_multiIndexedGeometry;
+}
+
+Geometry::MultiIndexedGeometry& GeometryData::getMultiIndexedGeometry() {
+    return m_multiIndexedGeometry;
+}
+
 const Utils::AttribManager& GeometryData::getAttribManager() const {
-    return m_vertexAttribs;
+    return m_multiIndexedGeometry.vertexAttribs();
 }
 
 Utils::AttribManager& GeometryData::getAttribManager() {
-    return m_vertexAttribs;
+    return m_multiIndexedGeometry.vertexAttribs();
+}
+
+template <typename T>
+inline Utils::Attrib<T>& GeometryData::getAttrib( const Geometry::MeshAttrib& name ) {
+    const auto& n = getAttribName( name );
+    auto h        = m_multiIndexedGeometry.addAttrib<T>( n );
+    auto& attrib  = m_multiIndexedGeometry.getAttrib( h );
+    return attrib;
+}
+
+template <typename V>
+inline void GeometryData::setAttribData( const Geometry::MeshAttrib& name,
+                                         const VectorArray<V>& attribDataList ) {
+    auto& attrib = getAttrib<V>( name );
+    auto& data   = attrib.getDataWithLock();
+    internal::copyData( attribDataList, data );
+    attrib.unlock();
```

Comment:
```suggestion
    getAttrib<V>( name ).setData(attribDataList);
```
if attribDataList can't be moved, maybe override setData

Reply:
I don't agree with this modification. I think it's better to lock and unlock the attrib because setData doesn't lock attrib, so if an other part try to write into it and the attrib was unlocked, two parts of the algorithms have write access.

Reply:
The mutex could be set either for lock, and setData ... so I don't see any difference here, expect 4 lines in place of 1 line.

Reply:
Yes, in fact